core/src/consensus/tendermint/types.rs

```diff
@@ -26,6 +26,42 @@ use super::message::VoteStep;
 pub type Height = usize;
 pub type View = usize;
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum InternalStepState {
+    Propose,
+    ProposeWaitBlockGeneration {
+        parent_hash: H256,
+    },
+    Prevote,
+    Precommit,
+    Commit,
+}
+
+impl InternalStepState {
+    pub fn to_step(&self) -> Step {
```

Comment:
Please implement `From<InternalStepState> for Step`

Reply:
I use this `to_step` function to use reference of `InternalStepState`.
If we need the `InternalStepState` value after converting it to `Step`, should we copy the `InternalStepState` value?

Reply:
I thought From is better since the copying cost of `InternalStepState` is cheap.
But now, implementing `to_step` also looks good.